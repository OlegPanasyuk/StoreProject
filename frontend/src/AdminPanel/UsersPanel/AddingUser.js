import React, { Component } from 'react';
import {
    Modal,
    Button,
    Form,
    Overlay,
    Tooltip
} from 'react-bootstrap';
import md5 from 'md5';
import PropTypes from 'prop-types';

// Redux
import { connect } from 'react-redux';
import {
    addErrorToState
} from '../../REDUX/actions/actionsErrors';

import {
    checkEmail,
    checkPassword,
    checkName
} from '../../utls/validators';

export class AddingUser extends Component {
    constructor(props) {
        super(props);
        this.attachRef = target => this.setState(state => ({
            tooltip: {
                ...state.tooltip,
                target
            }
        }));
        this.nameRef = React.createRef();
        this.emailRef = React.createRef();
        this.passwordRef = React.createRef();
        this.password2Ref = React.createRef();
        this.roleRef = React.createRef();
        this.sendUserToAdd = this.sendUserToAdd.bind(this);
        this.handleValid = this.handleValid.bind(this);
        this.state = {
            name: '',
            password: '',
            password2: '',
            email: '',
            tooltip: {
                target: null,
                message: '',
                show: false
            },
            emailValid: {
                valid: false,
                noValid: false,
                message: ''
            },
            passwordValid: {
                valid: false,
                noValid: false,
                message: ''
            },
            nameValid: {
                valid: false,
                noValid: false
            },
            equalPass: false
        };
    }


    handleValid(e) {
        if (e) e.preventDefault();
        const { name, value } = e.target;
        switch (name) {
        case this.nameRef.current.name: {
            const nameValid = checkName(value);
            this.setState({
                nameValid
            });
            break;
        }
        case this.emailRef.current.name: {
            const emailValid = checkEmail(value);
            this.setState({
                emailValid
            });
            break;
        }
        case this.passwordRef.current.name: {
            const { passwordValid, equalPass } = checkPassword(value, this.password2Ref.current.value);
            this.setState({
                passwordValid,
                equalPass
            });
            break;
        }
        case this.password2Ref.current.name: {
            const { passwordValid, equalPass } = checkPassword(value, this.passwordRef.current.value);
            this.setState({
                passwordValid,
                equalPass
            });
            break;
        }
        default: break;
        }
    }

    preValid() {
        let answ = false;
        const {
            emailValid, passwordValid, equalPass, nameValid
        } = this.state;
        if (emailValid.valid && passwordValid.valid && equalPass && nameValid.valid) {
            answ = true;
        } else if (this.passwordRef.current.value === '') {
            this.setState({
                passwordValid: {
                    valid: false,
                    noValid: true,
                    message: 'Enter password'
                }
            });
        } else if (!passwordValid.valid) {
            this.setState({
                passwordValid: {
                    valid: false,
                    noValid: true,
                    message: 'Must have length more then 8, have !@#$%^&*, letter uppercase and lowercase, number'
                }
            });
        } else if (!equalPass) {
            this.setState({
                passwordValid: {
                    valid: false,
                    noValid: true,
                    message: 'Passwords are not equal'
                }
            });
        } else {
            this.setState({
                passwordValid: {
                    valid: false,
                    noValid: false,
                    message: ''
                }
            });
        }
        return answ;
    }

    sendUserToAdd() {
        const storage = window.localStorage;
        const { addErrorToState, onHide, openPage } = this.props;
        if (this.preValid()) {
            if (fetch) {
                const myHeaders = new Headers();
                myHeaders.append('Authorization', `Bearer ${storage.getItem('Authorization')}`);
                myHeaders.append('Content-type', 'application/json');
                const body = {
                    name: this.nameRef.current.value,
                    password1: this.passwordRef.current.value,
                    password2: this.password2Ref.current.value,
                    email: this.emailRef.current.value,
                    role: this.roleRef.current.value
                };

                const myInit = {
                    method: 'POST',
                    headers: myHeaders,
                    body: JSON.stringify(body)
                };
                fetch(`${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}/users/new`, myInit)
                    .then((res) => {
                        if (res.status === 201) {
                            return res.json();
                        }
                        if (res.status === 401) {
                            res.json().then((data) => {
                                const d = new Date();
                                addErrorToState({
                                    id: md5(`${'Notification from AddingUser'}${d.valueOf()}`),
                                    level: 'Error',
                                    message: data.message
                                });
                            });
                        }
                        return null;
                    })
                    .then((data) => {
                        if (data) {
                            this.setState({
                                name: '',
                                password: '',
                                password2: '',
                                email: ''
                            });
                            onHide();
                            const d = new Date();
                            openPage(1);
                            addErrorToState({
                                id: md5(`${'Notification from AddingUser'}${d.valueOf()}`),
                                level: 'Success',
                                message: data.message
                            });
                        }
                    })
                    .catch((e) => {
                        this.setState(state => ({
                            tooltip: {
                                ...state.tooltip,
                                message: e,
                                show: true
                            }
                        }));
                    });
            }
        }
    }

    render() {
        const {
            tooltip,
            name,
            email,
            emailValid,
            password,
            password2,
            passwordValid,
            nameValid
        } = this.state;
        const { target, show, message } = tooltip;
        const { onHide } = this.props;
        return (
            <Modal
                show
                onHide={onHide}
                centered
            >
                <Modal.Header closeButton>
                    <Modal.Title>Adding User</Modal.Title>
                </Modal.Header>

                <Modal.Body>
                    <Form>
                        <Form.Group>
                            <Form.Label>
                                UserName
                            </Form.Label>
                            <Form.Control
                                ref={this.nameRef}
                                type='text'
                                value={name}
                                name='nameUser'
                                isValid={nameValid.valid}
                                isInvalid={nameValid.noValid}
                                onChange={(e) => {
                                    this.setState({
                                        name: this.nameRef.current.value
                                    });
                                    this.handleValid(e);
                                }}
                            />
                            <Form.Control.Feedback>{nameValid.message}</Form.Control.Feedback>
                            <Form.Control.Feedback type='invalid'>{nameValid.message}</Form.Control.Feedback>
                        </Form.Group>
                        <Form.Group>
                            <Form.Label>
                                Email
                            </Form.Label>
                            <Form.Control
                                ref={this.emailRef}
                                type='email'
                                value={email}
                                name='email'
                                isValid={emailValid.valid}
                                isInvalid={emailValid.noValid}
                                onChange={(e) => {
                                    this.setState({
                                        email: this.emailRef.current.value
                                    });
                                    this.handleValid(e);
                                }}
                            />
                            <Form.Control.Feedback>{emailValid.message}</Form.Control.Feedback>
                            <Form.Control.Feedback type='invalid'>{emailValid.message}</Form.Control.Feedback>
                        </Form.Group>
                        <Form.Group>
                            <Form.Label>
                                Password
                            </Form.Label>
                            <Form.Control
                                ref={this.passwordRef}
                                type='text'
                                value={password}
                                name='password1'
                                isValid={passwordValid.valid}
                                isInvalid={passwordValid.noValid}
                                onChange={(e) => {
                                    this.setState({
                                        password: this.passwordRef.current.value
                                    });
                                    this.handleValid(e);
                                }}
                            />
                            <Form.Control.Feedback>{passwordValid.message}</Form.Control.Feedback>
                            <Form.Control.Feedback type='invalid'>{passwordValid.message}</Form.Control.Feedback>
                        </Form.Group>
                        <Form.Group>
                            <Form.Label>
                                Repeat Password
                            </Form.Label>
                            <Form.Control
                                type='text'
                                isValid={passwordValid.valid}
                                isInvalid={passwordValid.noValid}
                                value={password2}
                                ref={this.password2Ref}
                                name='password2'
                                onChange={(e) => {
                                    this.setState({
                                        password2: this.password2Ref.current.value
                                    });
                                    this.handleValid(e);
                                }}
                            >
                            </Form.Control>
                        </Form.Group>

                        <Form.Group>
                            <Form.Label>
                                Role
                            </Form.Label>
                            <Form.Control
                                as='select'
                                ref={this.roleRef}
                                onChange={() => {}}
                            >
                                <option value='SuperAdmin'>SuperAdmin</option>
                                <option value='Admin'>Admin</option>
                                <option value='User'>User</option>
                                <option value='Customer'>Customer</option>
                            </Form.Control>
                        </Form.Group>
                    </Form>
                </Modal.Body>

                <Modal.Footer>
                    <Button
                        variant='light'
                        onClick={() => this.setState({
                            name: '',
                            password: '',
                            password2: '',
                            email: ''
                        })}
                    >
                        Clear
                    </Button>
                    <Button
                        variant='primary'
                        onClick={() => {
                            this.sendUserToAdd();
                        }}
                        ref={this.attachRef}
                    >
                        Save changes
                    </Button>
                    <Overlay target={target} show={show} placement='right'>
                        {props => (
                            <Tooltip id='overlay-example' {...props} show={show.toString()}>
                                {message}
                            </Tooltip>
                        )}
                    </Overlay>
                </Modal.Footer>
            </Modal>
        );
    }
}

AddingUser.propTypes = {
    onHide: PropTypes.func,
    addErrorToState: PropTypes.func,
    openPage: PropTypes.func
};

AddingUser.defaultProps = {
    onHide: () => {},
    addErrorToState: () => {},
    openPage: () => {}
};

export default connect(null, {
    addErrorToState
})(AddingUser);
