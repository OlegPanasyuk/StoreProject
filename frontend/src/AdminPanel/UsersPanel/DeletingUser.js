import React, { Component } from 'react';
import { Modal, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

export class DeletingUser extends Component {
    constructor(props) {
        super(props);
        this.sendUserToDelete = this.sendUserToDelete.bind(this);
    }

    sendUserToDelete() {
        let storage = window.localStorage;
        if (fetch) {
            let myHeaders = new Headers();
            myHeaders.append('Authorization', `Bearer ${storage.getItem('Authorization')}`);
            myHeaders.append("Content-type", 'application/json');
            let myInit = {
                method: 'DELETE',
                headers: myHeaders,
            };
            fetch(`${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}/users/${this.props.id}`, myInit)
                .then(res => {
                    return res.text();
                })
                .then(data => {
                    if (data) {
                        this.props.onHide();
                        // const d = new Date();
                        // this.props.addErrorToState({
                        //     id: md5(`${'Notification from Deleting'}${d.valueOf()}`),
                        //     level: 'Success',
                        //     message: 'User is deleted'
                        // });
                    }
                });
                
        }
    }

    render() {
        return (
            <Modal
                show={true}
                onHide={this.props.onHide}
                centered
            >
                <Modal.Header>
                    Delete this User?
                </Modal.Header>
                <Modal.Body className='d-flex justify-content-end'>
                    <Button 
                        className='ml-3'
                        onClick={this.sendUserToDelete}
                    >
                        Yes
                    </Button>
                    <Button 
                        variant='light'
                        className='ml-3'
                        onClick={this.props.onHide}
                    >
                        No
                    </Button>
                </Modal.Body>
            </Modal>
        );
    }
}

DeletingUser.propTypes = {
    id: PropTypes.number,
    onHide: PropTypes.func
};

export default DeletingUser;