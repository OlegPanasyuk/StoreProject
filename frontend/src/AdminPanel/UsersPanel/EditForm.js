import React, { Component } from 'react';
import {
    Modal, Button, Form, Overlay, Tooltip
} from 'react-bootstrap';
import md5 from 'md5';
import PropTypes from 'prop-types';

// Redux
import { connect } from 'react-redux';
import {
    addErrorToState
} from '../../REDUX/actions/actionsErrors';

export class EditForm extends Component {
    constructor(props) {
        super(props);
        this.attachRef = target => this.setState(state => ({
            tooltip: {
                ...state.tooltip,
                target
            }
        }));
        const { userToEdit } = this.props;
        const {
            username, password, email, role
        } = userToEdit;
        this.nameRef = React.createRef();
        this.emailRef = React.createRef();
        this.passwordRef = React.createRef();
        this.roleRef = React.createRef();
        this.sendUserToEdit = this.sendUserToEdit.bind(this);
        this.state = {
            username,
            password,
            email,
            role,
            tooltip: {
                target: null,
                message: '',
                show: false
            },
            emailValid: {
                valid: true,
                noValid: false,
                message: ''
            },
            passwordValid: {
                valid: true,
                noValid: false,
                message: ''
            },
            nameValid: {
                valid: true,
                noValid: false,
                message: ''
            }
        };
    }

    
    handleValid(e) {
        if (e) e.preventDefault();
        const { name, value } = e.target;
        const regEmail = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/i;
        const regPassWord = /(?=.*[0-9])(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z!@#$%^&*]{6,}/g;
        switch (name) {
        case this.nameRef.current.name: {
            if (value.length > 0) {
                this.setState({
                    nameValid: {
                        valid: true,
                        noValid: false,
                        message: ''
                    }
                });
            } else {
                this.setState({
                    nameValid: {
                        valid: false,
                        noValid: true,
                        message: 'Name must be not empty'
                    }
                });
            }
            break;
        }
        case this.emailRef.current.name: {
            if (regEmail.test(value)) {
                this.setState({
                    emailValid: {
                        valid: true,
                        noValid: false,
                        message: 'Email correct'
                    }
                });
            } else {
                this.setState({
                    emailValid: {
                        valid: false,
                        noValid: true,
                        message: 'Email must looks like: "email@domain.com"'
                    }
                });
            }
            break;
        }
        case this.passwordRef.current.name: {
            if (regPassWord.test(value)) {
                this.setState({
                    passwordValid: {
                        valid: true,
                        noValid: false,
                        message: 'Valid password'
                    }
                });
            } else {
                this.setState({
                    passwordValid: {
                        valid: false,
                        noValid: true,
                        message: 'Must have length more then 8, have !@#$%^&*, letter uppercase and lowercase, number'
                    }
                });
            }
            break;
        }
        default: break;
        }
    }

    preValid() {
        let answ = false;
        const { emailValid, nameValid, passwordValid } = this.state;
        if (emailValid.valid && nameValid.valid && passwordValid.valid) {
            answ = true;
        } else {
            answ = false;
        }
        return answ;
    }


    sendUserToEdit() {
        const storage = window.localStorage;
        const {
            username,
            password,
            email,
            role
        } = this.state;
        const {
            userToEdit, onHide, openPage, addErrorToState
        } = this.props;
        if (this.preValid()) {
            if (fetch) {
                const myHeaders = new Headers();
                myHeaders.append('Authorization', `Bearer ${storage.getItem('Authorization')}`);
                myHeaders.append('Content-type', 'application/json');
                const body = {
                    username,
                    password,
                    email,
                    role
                };

                const myInit = {
                    method: 'PUT',
                    headers: myHeaders,
                    body: JSON.stringify(body)
                };
                fetch(
                    `${
                        process.env.REACT_APP_API_HOST
                    }:${
                        process.env.REACT_APP_API_PORT
                    }/users/${
                        userToEdit.id
                    }`,
                    myInit
                )
                    .then(res => res.text())
                    .then((data) => {
                        if (data) {
                            this.setState({
                                username: '',
                                password: '',
                                email: '',
                                role: ''
                            });
                            onHide();
                            const d = new Date();
                            openPage(1);
                            addErrorToState({
                                id: md5(`${'Notification from AddingUser'}${d.valueOf()}`),
                                level: 'Success',
                                message: 'User is updated'
                            });
                        }
                    })
                    .catch((e) => {
                        this.setState(state => ({
                            tooltip: {
                                ...state.tooltip,
                                message: e,
                                show: true
                            }
                        }));
                    });
            }
        }
    }

    render() {
        const {
            tooltip, emailValid, nameValid, passwordValid
        } = this.state;
        const { target, show, message } = tooltip;
        const { onHide, userToEdit } = this.props;
        return (
            <Modal
                show
                onHide={onHide}
                centered
            >
                <Modal.Header closeButton>
                    <Modal.Title>Adding User</Modal.Title>
                </Modal.Header>

                <Modal.Body>
                    <Form>
                        <Form.Group>
                            <Form.Label>
                                UserName
                            </Form.Label>
                            <Form.Control
                                ref={this.nameRef}
                                type='text'
                                defaultValue={userToEdit.username}
                                name='nameUser'
                                isValid={nameValid.valid}
                                isInvalid={nameValid.noValid}
                                onChange={(e) => {
                                    this.setState({
                                        username: this.nameRef.current.value
                                    });
                                    this.handleValid(e);
                                }}
                            />
                            <Form.Control.Feedback>{nameValid.message}</Form.Control.Feedback>
                            <Form.Control.Feedback type='invalid'>{nameValid.message}</Form.Control.Feedback>
                        </Form.Group>
                        <Form.Group>
                            <Form.Label>
                                Email
                            </Form.Label>
                            <Form.Control
                                ref={this.emailRef}
                                type='text'
                                name='email'
                                defaultValue={userToEdit.email}
                                isValid={emailValid.valid}
                                isInvalid={emailValid.noValid}
                                onChange={(e) => {
                                    this.setState({
                                        email: this.emailRef.current.value
                                    });
                                    this.handleValid(e);
                                }}
                            />
                            <Form.Control.Feedback>{emailValid.message}</Form.Control.Feedback>
                            <Form.Control.Feedback type='invalid'>{emailValid.message}</Form.Control.Feedback>
                        </Form.Group>
                        <Form.Group>
                            <Form.Label>
                                Password
                            </Form.Label>
                            <Form.Control
                                ref={this.passwordRef}
                                type='text'
                                name='password'
                                defaultValue={userToEdit.password}
                                isValid={passwordValid.valid}
                                isInvalid={passwordValid.noValid}
                                onChange={(e) => {
                                    this.setState({
                                        password: this.passwordRef.current.value
                                    });
                                    this.handleValid(e);
                                }}
                            />
                            <Form.Control.Feedback>{passwordValid.message}</Form.Control.Feedback>
                            <Form.Control.Feedback type='invalid'>{passwordValid.message}</Form.Control.Feedback>
                        </Form.Group>

                        <Form.Group>
                            <Form.Label>
                                Role
                            </Form.Label>
                            <Form.Control
                                as='select'
                                defaultValue={userToEdit.role}
                                ref={this.roleRef}
                                onChange={() => {
                                    this.setState({
                                        role: this.roleRef.current.value
                                    });
                                }}
                            >
                                <option value='SuperAdmin'>SuperAdmin</option>
                                <option value='Admin'>Admin</option>
                                <option value='User'>User</option>
                                <option value='Customer'>Customer</option>
                            </Form.Control>
                        </Form.Group>
                    </Form>
                </Modal.Body>

                <Modal.Footer>
                    <Button
                        variant='light' onClick={() => {
                            this.setState({
                                username: userToEdit.username,
                                password: userToEdit.password,
                                email: userToEdit.email,
                                role: userToEdit.role
                            });
                        }}
                    >
                        Default
                    </Button>
                    <Button
                        variant='primary'
                        onClick={() => {
                            this.sendUserToEdit();
                        }}
                        ref={this.attachRef}
                    >
                        Save changes
                    </Button>
                    <Overlay target={target} show={show} placement='right'>
                        {props => (
                            <Tooltip id='overlay-example' {...props} show={show.toString()}>
                                {message}
                            </Tooltip>
                        )}
                    </Overlay>
                </Modal.Footer>
            </Modal>
        );
    }
}

EditForm.propTypes = {
    userToEdit: PropTypes.object,
    onHide: PropTypes.func,
    addErrorToState: PropTypes.func,
    openPage: PropTypes.func
};

EditForm.defaultProps = {
    userToEdit: {},
    onHide: () => null,
    addErrorToState: () => null,
    openPage: () => null
};

const mapStateToProps = state => ({
    userToEdit: state.adminPanel_usersPanel.userToEdit
});

export default connect(mapStateToProps, {
    addErrorToState
})(EditForm);
