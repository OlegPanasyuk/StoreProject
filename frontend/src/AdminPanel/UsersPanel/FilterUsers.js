import React, { Component } from 'react';
import { Form } from 'react-bootstrap';
import PropTypes from 'prop-types';

//Redux 
import { connect } from 'react-redux';
import { filterUsers } from '../../REDUX/adminPanel/actions/actionsUsersPanel';

export class FilterUsers extends Component {
    render() {
        return (
            <Form>
                <Form.Group>
                    <Form.Label>Name:</Form.Label>
                    <Form.Control
                        type='text'
                        placeholder='text input for search name'
                        onChange={
                            (e) => {
                                this.props.updateState(null, {
                                    nameSearch: e.target.value
                                });
                            }
                        }
                    />
                </Form.Group>

                <Form.Group>
                    <Form.Label>Role:</Form.Label>
                    <Form.Control
                        as='select'
                        onChange={(e) => {
                            this.props.updateState(null, {
                                role: e.target.value
                            });
                        }}
                    >
                        <option value=''>All</option>
                        <option value='SuperAdmin'>Super Admin</option>
                        <option value='Admin'>Admin</option>
                        <option value='Customer'>Customer</option>
                        <option value='User'>User</option>
                    </Form.Control>
                </Form.Group>

            </Form>

        );
    }
}

FilterUsers.propTypes = {
    updateState: PropTypes.func
};


export default connect(null, { filterUsers })(FilterUsers);