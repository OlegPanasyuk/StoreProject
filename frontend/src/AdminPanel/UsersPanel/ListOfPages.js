import React, { Component } from 'react';

//Components
import { Nav, Pagination } from 'react-bootstrap';

export class ListOfPages extends Component {
    render() {
        let { count, limit, activePage, openPage } = this.props;
        const numberOfPages = Math.ceil(count / limit);
        let listOfPages = [];
        for (let i = 1; i <= numberOfPages; i++) {
            listOfPages.push((
                <Nav.Item 
                    as={Pagination.Item} 
                    key={`nav-page-list-${i}`} 
                    active={i === activePage} 
                    onClick={() => openPage(i)}
                >
                    {i}
                </Nav.Item>
            ));
        }
        return (
            <Nav as={Pagination} size='sm' className='d-flex justify-content-center' >
                {listOfPages}
            </Nav>
        );
    }
}

export default ListOfPages;