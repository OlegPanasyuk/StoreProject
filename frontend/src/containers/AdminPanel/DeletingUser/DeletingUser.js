import React, { Component } from 'react';
import { Modal, Button } from 'react-bootstrap';
import md5 from 'md5';
import PropTypes from 'prop-types';

// Redux
import { connect } from 'react-redux';
import {
    addErrorToState
} from '../../../actions/actionsErrors';

export class DeletingUser extends Component {
    constructor(props) {
        super(props);
        this.sendUserToDelete = this.sendUserToDelete.bind(this);
    }

    sendUserToDelete() {
        const storage = window.localStorage;
        const {
            addErrorToState,
            onHide,
            openPage,
            id
        } = this.props;
        if (fetch) {
            const myHeaders = new Headers();
            myHeaders.append('Authorization', `Bearer ${storage.getItem('Authorization')}`);
            myHeaders.append('Content-type', 'application/json');
            const myInit = {
                method: 'DELETE',
                headers: myHeaders
            };
            fetch(`${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}/users/${id}`, myInit)
                .then((res) => {
                    if (res.status === 200) {
                        return res.text();
                    }
                    if (res.status === 400) {
                        res.json().then((e) => {
                            const d = new Date();
                            addErrorToState({
                                id: md5(`${'Notification from Deleting'}${d.valueOf()}`),
                                level: 'Error',
                                message: `${e}`
                            });
                        });
                    }
                    return null;
                })
                .then((data) => {
                    if (data) {
                        onHide();
                        const d = new Date();
                        openPage(1);
                        addErrorToState({
                            id: md5(`${'Notification from Deleting'}${d.valueOf()}`),
                            level: 'Success',
                            message: 'User is deleted'
                        });
                    }
                });
        }
    }

    render() {
        const { onHide } = this.props;
        return (
            <Modal
                show
                onHide={onHide}
                centered
            >
                <Modal.Header>
                    Delete this User?
                </Modal.Header>
                <Modal.Body className='d-flex justify-content-end'>
                    <Button
                        className='ml-3'
                        onClick={this.sendUserToDelete}
                    >
                        Yes
                    </Button>
                    <Button
                        variant='light'
                        className='ml-3'
                        onClick={onHide}
                    >
                        No
                    </Button>
                </Modal.Body>
            </Modal>
        );
    }
}

DeletingUser.propTypes = {
    id: PropTypes.number,
    onHide: PropTypes.func,
    addErrorToState: PropTypes.func,
    openPage: PropTypes.func
};

DeletingUser.defaultProps = {
    id: 0,
    onHide: () => {},
    addErrorToState: () => {},
    openPage: () => {}
};

export default connect(null, {
    addErrorToState
})(DeletingUser);
