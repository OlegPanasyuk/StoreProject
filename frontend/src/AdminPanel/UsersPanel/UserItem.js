import React, { Component } from 'react';
import { Card, Button, ButtonGroup } from 'react-bootstrap';
import PropTypes from 'prop-types';

//Redux 
import { connect } from 'react-redux';
import { editUser } from '../../REDUX/adminPanel/actions/actionsUsersPanel';

//Component
import DeletingUser from './DeletingUser';

export class UserItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            permissionToDelete: false
        };
    }

    render() {
        let { obj } = this.props;
        let deleteUser = null;
        if (this.state.permissionToDelete) {
            deleteUser = (
                <DeletingUser
                    id={obj.id}
                    onHide={() => {
                        this.setState({
                            permissionToDelete: false
                        });
                    }}
                    openPage={this.props.openPage}
                />
            );
        }
        return (
            <Card>
                {deleteUser}
                <Card.Header>
                    <ButtonGroup
                        className='mr-3'
                    >
                        <Button
                            variant='light'
                            onClick={() => {
                                this.props.editUser(obj);
                            }}
                        >
                            <i className="far fa-edit"></i>
                        </Button>
                        <Button
                            variant='light'
                            onClick={() => {
                                this.setState({
                                    permissionToDelete: true
                                });
                            }}
                        >
                            <i className="fas fa-trash-alt"></i>
                        </Button>
                    </ButtonGroup>
                </Card.Header>
                <Card.Body>
                    <Card.Title>{obj.username}</Card.Title>
                    <Card.Subtitle className="mb-2 text-muted">{obj.role}</Card.Subtitle>
                </Card.Body>
            </Card>
        );
    }
}

UserItem.propTypes = {
    obj: PropTypes.object,
    editUser: PropTypes.func,
    openPage: PropTypes.func
};

export default connect(null, {
    editUser
})(UserItem);