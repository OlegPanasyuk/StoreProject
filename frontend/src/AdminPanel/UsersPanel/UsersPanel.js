import React, { Component } from 'react';
import { Container, Row, Col, CardColumns, Button } from 'react-bootstrap';
import PropTypes from 'prop-types';

//Redux 
import { connect } from 'react-redux';
import {
    showUsers,
    editUserClose,
    filterUsers
} from '../../REDUX/adminPanel/actions/actionsUsersPanel';

//Components 
import UserItem from './UserItem';
import ListOfPages from './ListOfPages';
import FilterUsers from './FilterUsers';
import AddingUser from './AddingUser';
import EditUser from './EditForm';

export class UsersPanel extends Component {
    constructor(props) {
        super(props);
        this.openPage = this.openPage.bind(this);
        this.prepareSearchRow = this.prepareSearchRow.bind(this);
        this.updateState = this.updateState.bind(this);
        this.state = {
            showAddingModal: false
        };
    }

    prepareSearchRow() {
        let searchStr = [];
        let { filters } = this.props;
        Object.keys(filters).forEach((el) => {
            if ((el === 'nameSearch') && (filters[el] !== '')) {
                searchStr.push(`${el}=${filters[el]}`);
            } 
            if ((el === 'role') && (filters[el] !== '')) {
                searchStr.push(`${el}=${filters[el]}`);
            }
        });

        return searchStr.join('&');
    }

    updateState(e, obj, page = 1) {
        if (e) {
            e.preventDefault();
        }

        // I don't like it
        let a = new Promise((res, rej) => {
            const obj = {};
            this.props.filterUsers(obj);
            res(true);
            if (!obj) rej();
        });
        a.then(() => {
            this.openPage(page);
        }, () => {

        });
    }

    openPage(i) {
        let self = this;
        const storage = window.localStorage;
        if (fetch) {
            let searchStr = this.prepareSearchRow();
            let myHeaders = new Headers();
            myHeaders.append('Authorization', `Bearer ${storage.getItem('Authorization')}`);
            myHeaders.append("Content-type", 'application/json');
            let myInit = {
                method: 'GET',
                headers: myHeaders,
                cache: 'default'
            };
            fetch(
                `${
                    process.env.REACT_APP_API_HOST
                }:${
                    process.env.REACT_APP_API_PORT
                }/users/filter?page=${
                    i
                }&${
                    searchStr
                }`,
                myInit
            )
                .then((res) => {
                    self.props.filterUsers({activePage: i});
                    res.json().then(function (data) {
                        self.props.showUsers(data);
                    });
                });
        }
    }

    UNSAFE_componentWillMount() {
        const self = this;
        const storage = window.localStorage;
        if (fetch) {

            let myHeaders = new Headers();
            myHeaders.append('Authorization', `Bearer ${storage.getItem('Authorization')}`);
            myHeaders.append("Content-type", 'application/json');
            let myInit = {
                method: 'GET',
                headers: myHeaders,
                cache: 'default'
            };

            fetch(`${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}/users/filter?page=1`, myInit)
                .then((users) => {
                    return users.json();
                })
                .then((users) => {
                    self.props.showUsers(users);
                });
        }
    }

    render() {
        let addingModal = null;
        let editModal = null;
        let count = this.props.usersToShowCount;
        let {limit, activePage} = this.props.filters;
        if (this.state.showAddingModal) {
            addingModal = (
                <AddingUser
                    show={this.state.showAddingModal}
                    onHide={() => {
                        this.setState({
                            showAddingModal: false
                        });
                    }}
                />
            );
        }

        if (this.props.userToEdit.show) {
            editModal = (
                <EditUser
                    onHide={() => {
                        this.props.editUserClose();
                    }}
                />
            );
        }

        return (

            <Container>
                {addingModal}
                {editModal}
                <Row>
                    <Col>
                        <ListOfPages
                            count={count}
                            limit={limit}
                            activePage={activePage}
                            openPage={this.openPage}
                        />
                    </Col>
                </Row>
                <Row>
                    <Col className='col-3'>
                        <Button
                            onClick={() => {
                                this.setState({
                                    showAddingModal: true
                                });
                            }}
                        >
                            Add User
                        </Button>
                        <FilterUsers 
                            updateState = {this.updateState}
                        />
                    </Col>
                    <Col className='col-9'>
                        <CardColumns>
                            {(this.props.usersToShow) && this.props.usersToShow.map((el, i) => {
                                return (
                                    <UserItem key={`el${el.username}-${i}`} obj={el} />
                                );
                            })}
                        </CardColumns>
                    </Col>
                </Row>
            </Container>

        );
    }
}

UsersPanel.propTypes = {
    showUsers: PropTypes.func,
    filters: PropTypes.object,
    filterUsers: PropTypes.func,
    userToEdit: PropTypes.object,
    usersToShow: PropTypes.array,
    usersToShowCount: PropTypes.number,
    editUserClose: PropTypes.func
};

const mapStateToProps = (state) => {
    return {
        usersToShow: state.adminPanel_usersPanel.usersToShow.rows,
        usersToShowCount: state.adminPanel_usersPanel.usersToShow.count,
        userToEdit: state.adminPanel_usersPanel.userToEdit,
        filters: state.adminPanel_usersPanel.filters
    };
};

export default connect(mapStateToProps, {
    showUsers,
    editUserClose,
    filterUsers
})(UsersPanel);