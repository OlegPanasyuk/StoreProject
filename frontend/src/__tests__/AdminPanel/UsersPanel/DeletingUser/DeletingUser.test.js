import React from 'react';
import { DeletingUser } from '../../../../AdminPanel/UsersPanel/DeletingUser';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    id: 1,
    onHide: ()=>{}
}) {
    let component = shallow(<DeletingUser {...props} />);
    return { component, props };
}

describe('DeletingUser --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(DeletingUser.propTypes.id).toBe(PropTypes.number);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});
