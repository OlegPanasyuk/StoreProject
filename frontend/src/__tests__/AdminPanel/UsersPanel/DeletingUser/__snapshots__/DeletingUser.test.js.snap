// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletingUser --- Snapshot Component must be rendered 1`] = `
<Bootstrap(Modal)
  centered={true}
  onHide={[Function]}
  show={true}
>
  <ModalHeader
    closeButton={false}
    closeLabel="Close"
  >
    Delete this User?
  </ModalHeader>
  <ModalBody
    className="d-flex justify-content-end"
  >
    <Button
      active={false}
      className="ml-3"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Yes
    </Button>
    <Button
      active={false}
      className="ml-3"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      No
    </Button>
  </ModalBody>
</Bootstrap(Modal)>
`;
